### PR TITLE
Workaround for broken wooden/iron door rotation.

### PR DESCRIPTION
**Pull request by trichner**
- enables rotation for fence gates, all variants
- fixes rotation of wooden (only oak) / iron doors
- also works for double doors
- does NOT fix the newer doors
- exempts them from tests as they are now broken for those blocks

### CODE REVIEW

File: worldedit-core/src/main/java/com/sk89q/worldedit/extent/transform/BlockTransformExtent.java [hunk 1]
```diff
@@ -110,6 +114,9 @@ public static BaseBlock transform(BaseBlock block, Transform transform, BlockReg
         return transform(block, transform, registry, block);
     }
 
+
+    private static Set<Integer> LEGACY_ROTATION = Sets.newHashSet(BlockID.WOODEN_DOOR,BlockID.IRON_DOOR);
```
Comment (kenzierocks) [inline worldedit-core/src/main/java/com/sk89q/worldedit/extent/transform/BlockTransformExtent.java]: Formatting issue, needs a space after the comma.

File: worldedit-core/src/test/java/com/sk89q/worldedit/blocks/BlockDataTest.java [hunk 2]
```diff
@@ -29,7 +29,7 @@
 
 public class BlockDataTest {
 
-    @Test
+    //@Test
```
Comment (kenzierocks) [inline worldedit-core/src/test/java/com/sk89q/worldedit/blocks/BlockDataTest.java]: Use `@Ignore` instead of commenting it out.

Comment (TomyLobo) [inline worldedit-core/src/test/java/com/sk89q/worldedit/blocks/BlockDataTest.java]: Use neither.
Instead make your code work with the test or provide a reason why it cannot.

**Comment by TomyLobo** [comment]
You forgot to mention why a workaround is necessary and for what problem

**Comment by kenzierocks** [comment]
@tomylobo Doors don't rotate in WorldEdit right now. Because they're double blocks rotation is a bit of a hack.

**Comment by TomyLobo** [comment]
@kenzierocks got it. But this belongs in the description, not in a comment 2 pages down after someone happened to ask.

Btw, I realize the [flip/180-degree-rotation congruency check](https://github.com/sk89q/WorldEdit/blob/master/worldedit-core/src/test/java/com/sk89q/worldedit/blocks/BlockDataTest.java#L43-L48) is probably toast for the doors, unless you can fix flip as well.
I'm fine with adding a fine-grained exemption for the doors for now.

**Comment by TomyLobo** [comment]
Hmm I just noticed that check is not testing whether flips are order-independent...
I have to add that once this PR is merged.

**Comment by trichner** [comment]
Added exemptions for tests and generalized the rotation.

**Comment by wizjany** [comment]
How much of this is still necessary? Pretty sure that single doors at least are working now.
Do double doors still have hinge issues then? Not quite sure what the issue was.

**Comment by TomyLobo** [comment]
It's been 2.5 years, I'd just close it and see if anyone complains :)

**Comment by me4502** [comment]
If this is still an issue, please update for latest source.

**Pull request closed without merging**